### PR TITLE
Add DirectionTag class

### PR DESCRIPTION
**Pull request by MonicaBrandeis**
### Description:
This PR adds a direction tag to Atlas. 

The direction tag should be used with two specific conditions on a Node in OSM to provide more information. One usage is to combine  _highway=MINI_ROUNDABOUT_  for the directionality of mini-roundabout. Another usage is to combine  _highway=STOP_ or _highway=GIVE_WAY_ to represent the Stop/Yield Sign information in detail at the intersection.  Detailed combination and impacts are discussed below.

### Potential Impact:
1. The tag value _highway=MINI_ROUNDABOUT_ should be accompanied by _direction=CLOCKWISE_ or _direction=ANTICLOCKWISE_ to represent the directionality of the mini-roundabout. 

2. The tag value _highway=STOP_ or _highway=GIVE_WAY_  should be accompanied by _direction=FORWARD_ or _direction=BACKWARD_ to represent the which direction has to stop/give_way/yield and give priority to rest of traffic directions. 

Currently Atlas has no direction tag to deal with any of conditions mentioned above. Adding direction tag will improve the clarity for routing.  

Nothing downstream will be impacted as this is solely adding functionality.

### Unit Test Approach:

N/A

### Test Results:

N/A

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/tags/DirectionTag.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    FORWARD,
+    BACKWARD,
+    CLOCKWISE,
+    BOTH,
+    ANTICLOCKWISE;
+
+    @TagKey
+    public static final String KEY = "direction";
+}
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/tags/DirectionTag.java]: You should have a blank line at the end of the file.

**Review by Bentleysb — CHANGES_REQUESTED** [review]
Looks pretty good. Just the one comment on the code. The only other thing, and perhaps @danielduhh can confirm, your PR comment should have all the sections from [the template](https://github.com/osmlab/atlas/blob/dev/.github/PULL_REQUEST_TEMPLATE.md), even if they are just N/A or None.

**Review by danielduhh — CHANGES_REQUESTED** [review]
+1 one on new line at the of the file. Also, your PR description needs to be cleaned up

**Review by danielduhh — APPROVED** [review]

File: src/main/java/org/openstreetmap/atlas/tags/DirectionTag.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * @author MonicaBrandeis
+ */
+@Tag(taginfo = "https://taginfo.openstreetmap.org/keys/direction", osm = "https://wiki.openstreetmap.org/wiki/Key:direction")
+public enum DirectionTag
+{
+    FORWARD,
+    BACKWARD,
+    CLOCKWISE,
+    BOTH,
+    ANTICLOCKWISE;
```
Comment (savannahostrowski) [inline src/main/java/org/openstreetmap/atlas/tags/DirectionTag.java]: Nit: I'd move clockwise and anti-clockwise next to each other.

**Review by savannahostrowski — CHANGES_REQUESTED** [review]
Woo! Nice work @MonicaBrandeis!

It'd be awesome if you could also add an UNKNOWN enum to this class and then we might be able to refactor MalformedRoundabout to use this Enum entirely (That check uses a custom Enum with CLOCKWISE, COUNTERCLOCKWISE, UNKNOWN, and MULTIDIRECTIONAL).

**Comment by savannahostrowski** [comment]
For ease of viewing: [MalformedRoundabout](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java)

**Comment by savannahostrowski** [comment]
Spoke offline: changes to accommodate MalformedRoundabout are unnecessary.

**Review by savannahostrowski — APPROVED** [review]
LGTM - just that small nit.

**Review by Bentleysb — APPROVED** [review]

**Pull request closed without merging**